components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java

```diff
@@ -408,8 +414,62 @@ public Response applicationsApplicationIdKeysGet(String applicationId, MessageCo
     @Override
     public Response applicationsApplicationIdKeysKeyTypeGenerateTokenPost(String applicationId,
             String keyType, ApplicationTokenGenerateRequestDTO body, String ifMatch, MessageContext messageContext) {
-        // do some magic!
-        return Response.ok().entity("magic!").build();
+        try {
+            String username = RestApiUtil.getLoggedInUsername();
+            APIConsumer apiConsumer = RestApiUtil.getConsumer(username);
+            Application application = apiConsumer.getApplicationByUUID(applicationId);
+
+            if (application != null) {
+                if (RestAPIStoreUtils.isUserAccessAllowedForApplication(application)) {
+                    ApplicationKeyDTO appKey = getApplicationKeyByAppIDAndKeyType(applicationId, keyType);
+                    if (appKey != null) {
+                        String jsonInput = null;
+                        try {
+//                          verify that the provided validity period is a long
```

Comment:
Should we move the comment //s here to the beginning of the texts?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/6704